engine/classes/Elgg/Router.php

```diff
@@ -95,17 +95,24 @@ public function route(\Elgg\Http\Request $request) {
 			$handled = call_user_func($function, $segments, $identifier);
 		}
 
-		$output = ob_get_clean();
+		if ($handled === true) {
+			$output = ob_get_clean();
+		} else if (is_string($handled)) {
+			$output = $handled;
+		}
```

Comment:
The effective change here is to allow handlers to return a string. I'd instead allow an array to be returned, and pull out keys like you do in actions.